XCUITests/NavigationTest.swift

```diff
@@ -345,6 +345,84 @@ class NavigationTest: BaseTestCase {
+		XCTAssertTrue(app.menuItems["Select All"].exists)
+        XCTAssertTrue(app.menuItems["Select"].exists)
+        if UIPasteboard.general.string != nil {
+                XCTAssertTrue(app.menuItems["Paste"].exists)
+                XCTAssertTrue(app.menuItems["Paste & Go"].exists)
+            }
+
+        //Tap on Select option and make sure Copy, Cut, Paste, and Look Up are shown
+        app.menuItems["Select"].tap()
+        waitForExistence(app.menuItems["Copy"])
+
+        if UIDevice.current.userInterfaceIdiom == .pad {
+            XCTAssertTrue(app.menuItems["Copy"].exists)
+            XCTAssertTrue(app.menuItems["Cut"].exists)
+            XCTAssertTrue(app.menuItems["Look Up"].exists)
+            XCTAssertTrue(app.menuItems["Share…"].exists)
+            XCTAssertTrue(app.menuItems["Paste & Go"].exists)
+            if UIPasteboard.general.string != nil {
+                XCTAssertTrue(app.menuItems["Paste"].exists)
+            }
+        }
+        else {
+        XCTAssertTrue(app.menuItems["Copy"].exists)
+        XCTAssertTrue(app.menuItems["Cut"].exists)
+        XCTAssertTrue(app.menuItems["Look Up"].exists)
+        if UIPasteboard.general.string != nil {
+                XCTAssertTrue(app.menuItems["Paste"].exists)
+                XCTAssertTrue(app.menus.children(matching: .menuItem).element(boundBy: 4).exists)
+            }
+        }
+
+        //Go back from Select and redo the tap
+        app.textFields["address"].tap()
+		app.textFields["address"].press(forDuration: 2)
+
+    	//Tap on Select All option and make sure Copy, Cut, Paste, and Look Up are shown
+		app.menuItems["Select All"].tap()
+		waitForExistence(app.menuItems["Copy"])
+        if UIDevice.current.userInterfaceIdiom == .pad {
+            XCTAssertTrue(app.menuItems["Copy"].exists)
+            XCTAssertTrue(app.menuItems["Cut"].exists)
+            XCTAssertTrue(app.menuItems["Look Up"].exists)
+            XCTAssertTrue(app.menuItems["Share…"].exists)
+            //These options only show if clipboard is not nil
+            if UIPasteboard.general.string != nil {
+                XCTAssertTrue(app.menuItems["Paste"].exists)
+                XCTAssertTrue(app.menuItems["Paste & Go"].exists)
+            }
+        }
+        else {
+            XCTAssertTrue(app.menuItems["Copy"].exists)
+            XCTAssertTrue(app.menuItems["Cut"].exists)
+            XCTAssertTrue(app.menuItems["Look Up"].exists)
+            //XCTAssertTrue(app.menuItems["Share…"].exists)
+            //These options only show if clipboard is not nil
+            if UIPasteboard.general.string != nil {
+                XCTAssertTrue(app.menuItems["Paste"].exists)
+            }
+        }
+
+        //test whether the correct options Select, Select All, Paste, and Paste & Go show up after copying something
+        if UIPasteboard.general.string == nil {
+            app.menuItems["Copy"].tap()
+            testLongPressOnAddressBar()
```

Comment:
Calling the test here makes the test to be running forever...There has to be another way to check the clipboard and continue the test according to that without calling the test itself...

Reply:
It should only run once, right? Because it will copy something to pasteboard and then at the next pass of the function this will never execute. So if there is already something in the pasteboard, it will not execute- and if there's nothing in the pasteboard, it will only execute once. That's my intent at least xD

Reply:
So, something is not working because the test is executed forever and it has to be defined to run only once and pass or fail. Instead of calling the test itself, I would try to group what you want to execute in a different private function and call it from the test if the pasteboard is empty or not and so that function does what you want. But calling the test from itself could cause some issues...